tests/io/test_data_catalog.py

```diff
@@ -825,18 +826,44 @@ def test_dataset_not_in_catalog_when_no_pattern_match(
         with pytest.raises(DatasetError, match=re.escape(pattern)):
             catalog._get_dataset(dataset_name)
 
-    def test_dataset_pattern_ordering(
-        self, config_with_dataset_factories_only_patterns
-    ):
+    @pytest.mark.parametrize(
+        "unordered_list,ordered_list",
+        [
+            (
+                [
+                    "{default}",
+                    "{country}_companies",
+                    "{dataset_name}s",
+                    "{country}_{dataset}",
+                ],
+                [
+                    "{country}_companies",
+                    "{country}_{dataset}",
+                    "{dataset_name}s",
+                    "{default}",
+                ],
+            ),
+            (
+                [
+                    "{brand}_cars",
+                    "{brand}_scooters",
+                    "{brand}_planes",
+                ],
+                [
+                    "{brand}_scooters",
+                    "{brand}_planes",
+                    "{brand}_cars",
```

Comment:
Nit, I will reorder this to be the first test case since this is more straightforward, the first test case is the more complicated one.